lib/k8s/client/declarative.ex

```diff
@@ -0,0 +1,90 @@
+defmodule K8s.Client.Declarative do
+  @moduledoc """
+  Declarative functions for the K8s.Client module.
+  """
+
+  alias K8s.Client.Imperative
+  alias K8s.Client.Runner.Base
+
+  @last_applied_configuration_annotation "ex.kubernetes.io/last-applied-configuration"
+
+  @doc """
+  Like kubectl it applies the given resource to the cluster. Since the apply has to check whether the resource already
```

Comment:
I would be very conservative in comparisons to `kubectl`, particularly in phrasing that reads as "this implements `kubectl`'s behavior". "Inspired by" would be preferable.

Adjusting for linebreaks, etc.
```suggestion
  Inspired by `kubectl apply`, updates a given resource to the cluster, while recording the previous state in the annotation `(#{@last_applied_configuration_annotation})`. Since the apply has to check whether the resource already
```